### DISCUSSION TITLE
Server-Side Extension Usage

### DISCUSSION THREAD

**Discussion by CompuIves** [CodinGame/monaco-vscode-api#173] [Ideas]
Now that we have ways to run extensions in different contexts (main window, web worker), I wonder how feasible it is to run some of the extensions on the server. In that case we would run an extension host from a server, and it would communicate using a protocol (could be WebSocket, or allow users to custom implement it) with the editor. Is this something feasible with the current architecture?

**Comment by CGNonofr** [comment]
The ext-host type exists in vscode web. I don't see any reason why it wouldn't be possible

Out of curiosity, what is your use-case?

**Reply by CompuIves** [reply]
> The ext-host type exists in vscode web. I don't see any reason why it wouldn't be possible

Super interesting. I'm curious if it would be interesting to support the 'transfer" of the extensions. I think we could pre-download the extensions inside the VM.

> Out of curiosity, what is your use-case?

We're exploring enabling VSCode extensions in our Monaco-based environment, and letting users even install the extensions from the marketplace. In CodeSandbox, we're also running a server (VM) that can run the extensions, and this would enable us to run the server-side extensions as well. So this exploration could enable us to run any extension from the open-vsix store, while maintaining flexibility when it comes to maintaining the UI.

**Reply by CGNonofr** [reply]
Very interesting!

**Comment by CGNonofr** [comment]
Can you have a look at https://github.com/CodinGame/monaco-vscode-api/pull/184?

It allows to run extension on a server, it requires to have the extension files on the server and the manifest on the client.

It's demonstrated in the demo

BTW, it doesn't only bind the extension host but also all remote features (filesystem, file search, terminals...)

**Reply by CompuIves** [reply]
Absolutely! I'll take a look and do some testing today, thanks a lot for this. It looks very promising!
